client/beefy/src/round.rs

```diff
@@ -63,27 +57,37 @@ pub fn threshold(authorities: usize) -> usize {
 	authorities - faulty
 }
 
+#[derive(Debug, PartialEq)]
+pub enum VoteImportResult<B: Block> {
+	Ok,
+	RoundConcluded(SignedCommitment<NumberFor<B>, Signature>),
+	Equivocation, /* TODO: (EquivocationProof<NumberFor<B>, Public, Signature>) */
+	Invalid,
+	Stale,
+}
+
 /// Keeps track of all voting rounds (block numbers) within a session.
 /// Only round numbers > `best_done` are of interest, all others are considered stale.
 ///
 /// Does not do any validation on votes or signatures, layers above need to handle that (gossip).
 #[derive(Debug, Decode, Encode, PartialEq)]
-pub(crate) struct Rounds<Payload, B: Block> {
-	rounds: BTreeMap<(Payload, NumberFor<B>), RoundTracker>,
+pub(crate) struct Rounds<B: Block> {
+	rounds: BTreeMap<Commitment<NumberFor<B>>, RoundTracker>,
+	previous_votes: BTreeMap<(Public, NumberFor<B>), VoteMessage<NumberFor<B>, Public, Signature>>,
```

Comment:
Not that it matters much for this PR, but I think I would have instead stored this as `Map<NumberFor<B>, Set<VoteMessage<...>>`. The `VoteMessage` already includes the public key, and I was thinking there might be situations in the future where we want to grab all votes for a round. Also this doesn't allow us to store equivocated votes (but maybe you worked around that a different way).

Reply:
`previous_votes: Map<NumberFor<B>, Set<VoteMessage<...>>>,` would involve iterating over `Set` to look for a possibly equivocated `vote' != vote` (where `vote'` comes from same authority, but with a different commitment).

We could do `Map<NumberFor<B>, Map<Public, VoteMessage<...>>>` to keep same efficient lookup but still allow grabbing all votes for a round.

I say we leave it like this for now, and make the change when/if needed.